src/components/hooks/useNumberInput.ts

```diff
@@ -30,38 +36,43 @@ export default function useNumberInput(props: NumberInputProps) {
         [props]
     );
 
+    // Handle value increments and decrements
     const increment = useCallback(
         (amount) => {
             const newVal = props.value + amount;
             if (isNaN(newVal)) return;
             props.onChange(clamp(newVal, min, max));
+            if (props.onBlur) props.onBlur(newVal);
         },
         [props, min, max]
     );
 
+    // Increment and decrement value on keydown and keyup
```

Comment:
```suggestion
    // Increment and decrement value on ArrowDown and ArrowUp
```
keydown/keyup are key event types, not keys ^^